PukingPredator/Assets/Scripts/Puking.cs

```diff
@@ -75,8 +80,38 @@ private void GameInput_Puke()
             //...the velocity and if there was a rigid body then we can reduce the velocity
             //...while calculating it based on the mass?
             itemRb.AddForce(pukeVelocity * itemRb.mass, ForceMode.Impulse);
+
+            // -1 because player loses an item when they puke
+            if (pukeForce > MAX_PUKE_FORCE * PUKE_EXPLODE_THRESH)
+            {
+                KnockbackItemsInFrontofPlayer(pukeVelocity * pukeForce, pukeForce, itemRb);
+            }
         }
 
     }
 
+    public void KnockbackItemsInFrontofPlayer(Vector3 vec, float f, Rigidbody itemRB)
+    {
+
+        Vector3 halfExtents = new Vector3(1f, 0.5f, 0.5f); // Half the size of the box (x, y, z)
+
+        // BoxCast in the direction of the velocity
+        RaycastHit[] hits = Physics.BoxCastAll(transform.position + (transform.forward * 0.1f) + (transform.up * 0.6f), halfExtents, vec, Quaternion.identity, 3f);
+
+        foreach (RaycastHit hit in hits)
+        {
+            if (!hit.collider.CompareTag(GameTag.player))
+            {
+                Rigidbody hitRB = hit.collider.GetComponent<Rigidbody>();
+                PhysicsBehaviour pb = hit.collider.GetComponent<PhysicsBehaviour>();
+                if (hitRB != null && pb != null && hitRB != itemRB)
```

Comment:
extract the check into smaller guard clauses for readability and also to make RBs respond even without a physics behaviour. ie
var hitRB = ...;
if (hitRB == null || hitRB == itemRB) { continue; }
var pb = ...;
if (pb != null) { rb.EnablePhysics(); }
hitRB.AddExplosive...;